### ISSUE TITLE
Typeorm - extend/merge next-auth methods in graphQL resolvers 

### ISSUE THREAD

**Issue by trompx** [nextauthjs/next-auth#420]
**Your question**
Hello @iaincollins,
I just checked the new released doc, it is now clear how to extend models as seen [here](https://next-auth-docs-git-v3.iaincollins.vercel.app/tutorials/typeorm-custom-models), thanks!
Available methods are also now listed [here](https://next-auth-docs-git-v3.iaincollins.vercel.app/tutorials/creating-a-database-adapter) but I am wondering how I could extend or merge those methods in my resolvers?

**What are you trying to do**
I'd like to integrate and extend next-auth methods to register, login and do some onboarding stuff after account creation, directly inside my current graphq resolvers (see below). I can't picture how to integrate next-auth and replace my **register** and **login** methods with next-auth ones as they are supposed to be declared in the adapter. Do you have a small example of how next-auth REST API can integrate with this Graphql API setup?
``` ts
import { Resolver, Query, Mutation, Arg, Authorized } from "type-graphql"
import { User } from "./user.entity"
import { UserService } from "./user.service"
import { createToken } from "@api/lib/jwt"
import { LoginInput } from "./inputs/login.input"
import { RegisterInput } from "./inputs/register.input"
import { UserAuthResponse } from "./user.response"
import { ConnectedUser } from "@context/connectedUser"

@Resolver(() => User)
export class UserResolver {
  constructor(private readonly userService: UserService) {}

  @Authorized()
  @Query(() => User, { nullable: true })
  async me(@ConnectedUser() user: User): Promise<User> {
    return user
  }

  @Mutation(() => UserAuthResponse)
  async register(@Arg("data") data: RegisterInput): Promise<UserAuthResponse> {
    const user = await this.userService.create(data)
    const token = await createToken(user.id)
    return { user, token }
  }

  @Mutation(() => UserAuthResponse)
  async login(@Arg("data") data: LoginInput): Promise<UserAuthResponse> {
    const user = await this.userService.login(data)
    const token = await createToken(user.id)
    return { user, token }
  }
}

```

**Documentation feedback**
*Documentation refers to searching through [online documentation](https://next-auth.js.org), code comments and issue history. The example project refers to [next-auth-example](https://github.com/iaincollins/next-auth-example).*

* [ ] Found the documentation helpful
* [x] Found documentation but was incomplete
* [x] Could not find relevant documentation
* [ ] Found the example project helpful
* [ ] Did not find the example project helpful

**Comment by iaincollins** [comment]
I'm so glad it was helpful!  @tmayr did an amazing job fixing a show stopping bug with custom models and on that tutorial.



I am not sure how to do what you are asking, but this seems like a good question someone else might be able to help with.

PS: You might also be tangentially interested in PR #397 (which wasn't merged, but was resolved in #415); it doesn't directly address your question but I mention it in case you find it relevant.

**Comment by amosbastian** [comment]
@trompx did you ever figure this out?
